src/battle_main.c

```diff
@@ -1984,7 +1984,10 @@ u8 CreateNPCTrainerPartyFromTrainer(struct Pokemon *party, const struct Trainer
             else if (partyData[i].gender == TRAINER_MON_FEMALE)
                 personalityValue = (personalityValue & 0xFFFFFF00) | GeneratePersonalityForGender(MON_FEMALE, partyData[i].species);
             if (partyData[i].nature != 0)
-                ModifyPersonalityForNature(&personalityValue, partyData[i].nature - 1);
+            {
+                while (GetNatureFromPersonality(personalityValue) != partyData[i].nature - 1)
+                    ModifyPersonalityForNature(&personalityValue, partyData[i].nature - 1);
```

Comment:
I worry about this `while`, and I'm getting different amount of rerolls based on code changes
![image](https://github.com/rh-hideout/pokeemerald-expansion/assets/2904965/8ca5db5f-956e-4087-a163-2091b163d261)

Reply:
I think the function modifying nature isn't working like it should and that's why it was failing before.

Reply:
Then we should fix that instead of doing a `while` I feel ^^;